sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -690,7 +690,15 @@ object PushPredicateThroughAggregate extends Rule[LogicalPlan] with PredicateHel
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case filter @ Filter(condition,
         aggregate @ Aggregate(groupingExpressions, aggregateExpressions, grandChild)) =>
-      val (pushDown, stayUp) = splitConjunctivePredicates(condition).partition {
+
+      // Create a map of Alias for grouping keys or literals
+      val aliasMap = AttributeMap(aggregateExpressions.collect {
+        case a: Alias if groupingExpressions.contains(a.child) || a.child.foldable =>
```

Comment:
This doesn't work if a grouping expression is inside another expression, for example, `key + 1 as k`, after remove the alias, `groupingExpressions` doesn't contain `key + 1` and we will fail to push down it.

I think we don't need the `if` here, we have `conjunct.references subsetOf AttributeSet(groupingExpressions)` below to decide whether to push a condition or not.